tests/integration_tests/datasources/test_oci_networking.py

```diff
@@ -0,0 +1,158 @@
+    tmpdir,
+    configure_secondary_nics: bool = False,
+):
+    assert client.execute("rm -f /run/initramfs/open-iscsi.interface").ok
+    # Force network config
+    assert client.execute(
+        "rm -f /etc/cloud/cloud.cfg.d/99-disable-network-config.cfg"
+    ).ok
+
+    cfg = tmpdir.join("01_oracle_datasource.cfg")
+    with open(cfg, "w") as f:
+        f.write(
+            DS_CFG.format(configure_secondary_nics=configure_secondary_nics)
+        )
+    client.push_file(cfg, "/etc/cloud/cloud.cfg.d/01_oracle_datasource.cfg")
+
+    client.execute("cloud-init clean --logs")
+    client.restart()
+
+
+def extract_interface_names(network_config: dict) -> Set[str]:
+    if network_config["version"] == 1:
+        interfaces = map(lambda conf: conf["name"], network_config["config"])
+    elif network_config["version"] == 2:
+        interfaces = network_config["ethernets"].keys()
+    else:
+        raise NotImplementedError(
+            f'Implement me for version={network_config["version"]}'
+        )
+    return set(interfaces)
+
+
+@pytest.mark.oci
+def test_oci_networking_iscsi_instance(client: IntegrationInstance, tmpdir):
+    customize_environment(client, tmpdir, configure_secondary_nics=False)
+    result_net_files = client.execute("ls /run/net-*.conf")
+    assert result_net_files.ok, "No net files found under /run"
+
+    log = client.read_from_file("/var/log/cloud-init.log")
+    verify_clean_log(log)
+
+    assert (
+        "opc/v2/vnics/" not in log
+    ), "vnic data was fetched and it should no have been"
+
+    netplan_yaml = client.read_from_file("/etc/netplan/50-cloud-init.yaml")
+    netplan_cfg = yaml.safe_load(netplan_yaml)
+    configured_interfaces = extract_interface_names(netplan_cfg["network"])
+    assert 1 <= len(
+        configured_interfaces
+    ), "Expected at least 1 primary network configuration."
+
+    expected_interfaces = set(
+        re.findall(r"/run/net-(.+)\.conf", result_net_files.stdout)
+    )
+    for expected_interface in expected_interfaces:
+        assert (
+            f"Reading from /run/net-{expected_interface}.conf" in log
+        ), "Expected {expected_interface} not found in: {log}"
+
+    not_found_interfaces = expected_interfaces.difference(
+        configured_interfaces
+    )
+    assert not not_found_interfaces, (
+        f"Interfaces, {not_found_interfaces}, expected to be configured in"
+        f" {netplan_cfg['network']}"
+    )
+    assert client.execute("ping -c 2 canonical.com").ok
+
+
+@pytest.fixture(scope="function")
+def client_with_secondary_vnic(
+    session_cloud: IntegrationCloud,
+) -> Iterator[IntegrationInstance]:
+    """Create an instance client and attach a temporary vnic.
+
+    Note: It assumes the associated compartment has at least one subnet and
+    creates the vnic in the first encountered subnet.
+    """
+    with session_cloud.launch() as client:
+        compute_client = session_cloud.cloud_instance.compute_client
+
+        subnet_id = (
```

Comment:
Great work here, but do you think this could be added to pycloudlib basically overriding
[add_network_interface](https://github.com/canonical/pycloudlib/blob/main/pycloudlib/instance.py#L173) for Oracle ?

If that's too big an ask right now, I think we can keep this as-is, but it'd be good to add it after this PR merges.

Reply:
PR created here: https://github.com/canonical/pycloudlib/pull/203

Thanks for pointing out this. I mark this PR as draft until we integrate the new `pycloudlib`'s version.

Reply:
Done.